sklearn/preprocessing/data.py

```diff
@@ -1946,3 +1957,149 @@ def transform(self, X):
+        The indices of the features to be transformed
+    lambdas_ : array of float, shape (n_transformed_features,)
+        The parameters of the BoxCox transform for the selected features.
+    n_features_ : int
+        Number of features in input during fit
+
+    Notes
+    -----
+    The Box-Cox transform is given by::
+
+        y = (x ** lmbda - 1.) / lmbda,  for lmbda > 0
+            log(x),                     for lmbda = 0
+
+    ``boxcox`` requires the input data to be positive.
+
+    """
+    def __init__(self, feature_indices=None, n_jobs=1, copy=True):
+        self.feature_indices = feature_indices
+        self.n_jobs = n_jobs
+        self.copy = copy
+
+    def fit(self, X, y=None):
+        """Estimate lambda for each feature to maximise log-likelihood
+
+        Parameters
+        ----------
+        X : array-like, shape [n_samples, n_features]
+            The data to fit by apply boxcox transform,
+            to each of the features and learn the lambda.
+
+        """
+        X = check_array(X, ensure_2d=True, dtype=FLOAT_DTYPES)
+        self.n_features_ = X.shape[1]
+        if self.feature_indices is None:
+            self.feature_indices_ = np.arange(self.n_features_)
+        else:
+            self.feature_indices_ = np.copy(self.feature_indices)
+            if self.feature_indices_.dtype == np.bool:
+                self.feature_indices_ = np.where(self.feature_indices_)[0]
+        if any(np.any(X[:, self.feature_indices_] <= 0, axis=0)):
+            raise ValueError("BoxCox transform can only be applied "
+                             "on positive data")
+        out = Parallel(n_jobs=self.n_jobs)(delayed(_boxcox)(X[:, i],
+                                           lambda_x=None)
+                                           for i in self.feature_indices_)
+        self.lambdas_ = np.array([o[1] for o in out])
+        return self
+
+    def transform(self, X, y=None):
+        """Transform each feature using the lambdas evaluated during fit time
+
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, n_features)
+            The data to apply boxcox transform.
+
+        Returns
+        -------
+        X_tr : array-like, shape (n_samples, n_features)
+            The transformed data.
+        """
+        X = check_array(X, ensure_2d=True, dtype=FLOAT_DTYPES, copy=self.copy)
+        if any(np.any(X[:, self.feature_indices_] <= 0, axis=0)):
+            raise ValueError("BoxCox transform can only be applied "
+                             "on positive data")
+        if X.shape[1] != self.n_features_:
+            raise ValueError("X has a different shape than during fitting.")
+        X_tr = _transform_selected(X, self._transform, self.feature_indices_,
+                                   copy=False, order=True)
+        return X_tr
+
+    def _transform(self, X):
+        outputs = Parallel(n_jobs=self.n_jobs)(
+            delayed(_boxcox)(X[:, i], self.lambdas_[i])
```

Comment:
This is very unlikely to be much faster for Parallel. _boxcox should be a fast operation where lambda is given.

(For the case where all features are being transformed and all lambdas are > 0, you can also just do
```
return (X ** self.lambdas_ - 1) / lambdas_
```
I'm not certain whether handling this special case would provide any benefit.